File: csharp/Day01.cs
using aoc2022.Utils;

namespace aoc2022;

public class Day01
{
    private List<int> _input;

    public Day01()
    {
        var strInput = FileUtils.DayInputToString("01");
        _input = strInput
            .Split("\n")
            .Aggregate(new List<int> { 0 }, (results, s) =>
            {
                if (s == "")
                {
                    results.Add(0);
                }
                else
                {
                    results[^1] += int.Parse(s);
                }

                return results;
            });
    }

    public int Part1() =>
        _input
            .Max();


    public int Part2() =>
        _input
            .OrderByDescending(i => i)
            .Take(3)
            .Sum();

}

File: python/day01.py
input = open("../input/day01.txt", "r").read().splitlines()

elves = []

total = 0
for line in input:
    if line == "":
        elves.append(total)
        total = 0
    else:
        total += int(line)

# part 1
print(max(elves))

# part 2
elves.sort()
print(sum(elves[-3:]))
